lib/publiccloud/provider.pm

```diff
@@ -708,19 +708,31 @@ sub terraform_param_tags
 =head2 get_terraform_output
 
 Query the terraform data structure in json format.
+Automatic search and set valid terraform directory on helper image.
 Input: <jq-query-format> string; <empty> = no query then full output of data structure.
 E.g: to get the VM instance name from json data structure, the call is: 
     get_terraform_output(".vm_name.value[0]");
 To get the complete output structure, the call is:
-    get_terraform_output();
+    get_terraform_output(".");
 
 =cut
 
 sub get_terraform_output {
     my ($self, $jq_query) = @_;
-    my $res = script_output("terraform output -no-color -json | jq -r '$jq_query' 2>/dev/null", proceed_on_failure => 1);
-    # jq 'null' shall return empty
-    return $res unless ($res =~ /^null$/);
+    # cuurent dir saved
+    my $dir = script_output("pwd");
+    # terraform state file search first matching
+    my $cd = script_output("(find ~ -name *.tfstate -print | head -1 | xargs dirname) 2>/dev/null");
```

Comment:
Let's escape the asterisk and avoid the pipeline like this:

```suggestion
    my $cd = script_output("find ~ -name \*.tfstate -exec dirname {} -quit 2>/dev/null");
```

Reply:
First, let we keep external eveloping (...), used to send every pipeline error to stderr. If removed, the $cd would receive `find` error ouput.

Reply:
> First, let we keep external eveloping (...), used to send every pipeline error to stderr. If removed, the $cd would receive `find` error ouput.

The parentheses are only needed if we have multiple commands in a pipeline.  In this case we only have one.

Reply:
Why escaped `\*` ?

Reply:
No pipeline, applying your suggestion :thinking:

Reply:
> No pipeline, applying your suggestion 🤔 After -exec also `\;` neded.

`+` is also valid in this case and avoids 3 extra chars as you have escape the backslash.